File: examples/lua53/server/protocols/AckSceneEnter.lua
local M = {}
local moduleName = ...
_G[moduleName] = M
setmetatable(M, {__index = _G})
local _ENV = M


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckSceneEnter})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.player = nil
end


function encode(self)
	pack = packet.new()
	pack:writeBytes(self.player:getBytes())
	return pack:encode(Msg.P_ACK_SCENE_ENTER)
end


function setPlayer(self, player)
	self.player = player
end

File: examples/golang/server/proto/ack_scene_enter.go
package proto

import (
	"packet"
)

type AckSceneEnter struct {
	Player                   *MsgScenePlayer
}

func (this *AckSceneEnter) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteBytes(this.Player.Encode())

	return pack.Encode(P_ACK_SCENE_ENTER)
}
